apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PooledSession.cs

```diff
@@ -395,29 +395,16 @@ private PooledSession AfterReset()
         }
 
         /// <summary>
-        /// Returns a PooledSession for the same session as this one with the given transaction options and a newly created transaction.
+        /// Returns a PooledSession for the same session as this with the given <paramref name="transactionOptions"/> and no associated transaction.
         /// The refresh and eviction times are the same as this instance.
         /// This instance will be <see cref="MarkAsDisposed"/> to ensure that all operations with the underlying session are done through the new instance.
         /// </summary>
-        internal async Task<PooledSession> WithFreshTransactionAsync(TransactionOptions transactionOptions, CancellationToken cancellationToken)
+        internal PooledSession WithTransactionOptions(TransactionOptions transactionOptions, bool singleUseTransaction)
         {
             CheckNotDisposed();
             GaxPreconditions.CheckNotNull(transactionOptions, nameof(transactionOptions));
-            var transaction = await BeginTransactionAsync(cancellationToken).ConfigureAwait(false);
-            return WithTransaction(transaction.Id, transactionOptions, singleUseTransaction:false, transaction.ReadTimestamp);
-
-            Task<Transaction> BeginTransactionAsync(CancellationToken cancellationToken)
-            {
-                var request = new BeginTransactionRequest
-                {
-                    Options = transactionOptions,
-                    SessionAsSessionName = SessionName,
-                };
-                var callSettings = Client.Settings.BeginTransactionSettings
-                    .WithExpiration(Expiration.FromTimeout(_pool.Options.Timeout))
-                    .WithCancellationToken(cancellationToken);
-                return RecordSuccessAndExpiredSessions(Client.BeginTransactionAsync(request, callSettings));
-            }
+            MarkAsDisposed();
```

Comment:
This prevents it from being reused, right? Was it a bug that we didn't have that before?